packages/vscode-extension/src/project/project.ts

```diff
@@ -60,17 +60,28 @@ export class Project implements Disposable, MetroDelegate, ProjectInterface {
     selectedDevice: undefined,
   };
 
-  private deviceSettings: DeviceSettings = extensionContext.workspaceState.get(
-    DEVICE_SETTINGS_KEY
-  ) ?? {
-    appearance: "dark",
-    contentSize: "normal",
-    location: {
-      latitude: 50.048653,
-      longitude: 19.965474,
-      isDisabled: true,
-    },
-  };
+  getSafeDeviceSetting(): DeviceSettings {
+    const savedSettings: DeviceSettings | undefined =
+      extensionContext.workspaceState.get(DEVICE_SETTINGS_KEY);
+    if (!savedSettings) {
+      return {
+        appearance: "dark",
+        contentSize: "normal",
+        location: {
+          latitude: 50.048653,
+          longitude: 19.965474,
+          isDisabled: true,
+        },
+        showFrame: true,
```

Comment:
I think settings like that should be located tin WorkspaceConfiguration, similarly to how "Panel Location" is implemented. We could possibly  even just relay on vscode internal ui for "User Settings".